### ISSUE TITLE
Request termination plugin ontop of route not working

### ISSUE THREAD

**Issue by jeremyjpj0916** [Kong/kong#3375]
### Summary

I was looking into some advice by Tieske here https://discuss.konghq.com/t/kong-public-status-endpoint/762

So I created a Service, it had properties like such:

![image](https://user-images.githubusercontent.com/31913027/38291187-85defece-37ac-11e8-87e6-71e2a98fb23f.png)

And here is the DB result in C*:
![image](https://user-images.githubusercontent.com/31913027/38291219-a94e6994-37ac-11e8-9dfb-c4196ac1803a.png)

Then create a route:
![image](https://user-images.githubusercontent.com/31913027/38291236-cbb95dc2-37ac-11e8-9dcb-29615a1608f2.png)

Looks like so in the DB C*:
![image](https://user-images.githubusercontent.com/31913027/38291258-e74e1d2a-37ac-11e8-962c-7a7b7d638250.png)

Then Apply the Request Termination plugin to the route, json will come out like so:
![image](https://user-images.githubusercontent.com/31913027/38291293-0f27d9e4-37ad-11e8-98a8-51a691687763.png)

But then call the Proxy and still see it hitting stub status and has proxy behavior before the plugin was added:

![image](https://user-images.githubusercontent.com/31913027/38291359-5cd26682-37ad-11e8-8d29-ee447ea6e78f.png)

Also I am running it in diff places/browsers, its not browser cache related I checked that.

### Steps To Reproduce

1. Create Service pointing to admin api /status localhost style
2. Create Route to expose that endpoint 
3. Apply Request termination plugin on top of said route
4. Call the route, notice request termination plugin not kicking in and its just displaying the usual /status info from admin API call.

Note I did call the proxy URL without the plugin some too prior to adding the request-termination plugin, not that that should matter as its whole purpose is to be able to abruptly stop all traffic to backend endpoint at once . 

### Additional Details & Logs

- Kong version 0.13.0
- C* : Cassandra 3.11.2

**Comment by jeremyjpj0916** [comment]
Okay, I have some further info to reveal, maybe this will clarify how it could happen.

1. I had an API resource prior that had the same path and upstream URL with no plugin on top of it. I deleted said resource after I finished configuring the Service/Route resources.

2. The behavior of the proxy endpoint even after deleting the API resource and verifying it was not in the db showed behavior of the API resource as opposed to Kong finding the API resource removed and a fresh cache of the new Route/Service references with the plugin enabled ontop of them.

3. AFTER bouncing the pods kong was running on and starting it up fresh again only then did the plugin begin to kick in.

Would this be expected behavior? Should the DELETE of an API resource not purge it from Kong's cache? And then a subsequent call to the Service/Route cache the proxy info + plugin info needed?

-Jeremy
